File: python/1513-number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        total: int = 0
        contiguous: int = 0
        for i in range(0, len(s)):
            if s[i] == '1':
                contiguous += 1
            else:
                total += sum(range(1, contiguous + 1))
                contiguous = 0

        if contiguous != 0:
            total += sum(range(1, contiguous + 1))

        return total % (10 ** 9 + 7)

File: go/1513-number-of-substrings-with-only-1s.go
package solution

import (
	"math"
)

func numSub(s string) int {
	var total int = 0
	var contiguous int = 0

	for _, ch := range s {
		if string(ch) == "1" {
			contiguous++
		} else {
			for i := 1; i <= contiguous; i++ {
				total += i
			}
			contiguous = 0
		}
	}
	if contiguous != 0 {
		for i := 1; i <= contiguous; i++ {
			total += i
		}
	}

	return total % int(math.Pow(10, 9)+7)
}
